### PR TITLE
Dokka initial setup

### PR DESCRIPTION
**Pull request by RingerJK**
## Description

Initial Dokka setup 

- [ ] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

**Comment by korshaknn** [comment]
do we need to setup dokka for `libnavigation-util` and `libtesting`?

**Comment by RingerJK** [comment]
@korshaknn `util` module is used as internal dependency and we don't need public docs for it. 
 `libtesting` is as well. Correct me if I'm wrong.

**Comment by RingerJK** [comment]
@LukasPaczos added

File: Makefile [hunk 1]
```diff
@@ -21,6 +21,17 @@ javadoc:
 	./gradlew :libandroid-navigation:javadocrelease
 	./gradlew :libandroid-navigation-ui:javadocrelease
 
+javadoc-dokka:
+    ./gradlew :libnavigation-base:dokka
```
Comment (LukasPaczos) [inline Makefile]: @RingerJK you need to use tabs instead of spaces :sparkles:

File: Makefile [hunk 2]
```diff
@@ -21,6 +21,17 @@ javadoc:
...
 
+javadoc-dokka:
+	./gradlew :libnavigation-base:dokka
+	./gradlew :libnavigator:dokka
+	./gradlew :libdirections-onboard:dokka
+	./gradlew :libdirections-offboard:dokka
+	./gradlew :libdirections-hybrid:dokka
+	./gradlew :libnavigation-metrics:dokka
+	./gradlew :libtrip-notification:dokka
+	./gradlew :liblogger:dokka
```
Comment (LukasPaczos) [inline Makefile]: This needs to be removed.

**Comment by RingerJK** [comment]
@Guardiola31337 I can share my experience from Vision project: we create javadocs only. You can have a look how it looks like at https://docs.mapbox.com/android/vision/overview/

**Comment by LukasPaczos** [comment]
I'd prefer releasing a KDoc spec, that's what we decided on Maps SDK and it looks :sparkles:. Anything that we build today is not going to land on the website yet, so let's merge what we've got and we can iterate.

**Comment by RingerJK** [comment]
@LukasPaczos could you share a link to KDoc of Maps? I found javadoc only

**Comment by Guardiola31337** [comment]
> I'd prefer releasing a KDoc spec, that's what we decided on Maps SDK and it looks ✨

Yeah, totally agree ✨

> Anything that we build today is not going to land on the website yet, so let's merge what we've got and we can iterate.

Of course, not blocking the PR by any means! I only wanted to chime in so we discuss this as a team before releasing `1.0` 🚀

**Comment by Guardiola31337** [comment]
> I can share my experience from Vision project: we create javadocs only. You can have a look how it looks like at https://docs.mapbox.com/android/vision/overview/

To give a concrete example from Vision, [`VisionManager`](https://github.com/mapbox/mapbox-vision/blob/dev/SDK/Platforms/Android/sdk/src/main/java/com/mapbox/vision/VisionManager.kt#L66) is an `object` and Javadoc generates [`INSTANCE`](https://docs.mapbox.com/android/api/vision/vision/0.10.1/com/mapbox/vision/VisionManager.html#INSTANCE) that Kotlin users don't need.

My point is that we should have separate pages for Kotlin and Java users, as we'd get that for free because Dokka can generate documentation in multiple formats including Markdown and HTML that we can use in our docs site.

**Comment by langsmith** [comment]
> @LukasPaczos could you share a link to KDoc of Maps? I found javadoc only

Fyi @RingerJK  , https://github.com/mapbox/mapbox-maps-android/pull/58#issue-324436150 shows what the Dokka kdocs look like.

**Comment by RingerJK** [comment]
@langsmith thanks! looks nice 👍

**Pull request merged**